modules/disko/disko-ab-partitions.nix

```diff
@@ -0,0 +1,161 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# This partition scheme contains three common partitions and ZFS pool.
+# Some partitions are duplicated for the future AB SWupdate implementation.
+#
+# First three partitions are related to the boot process:
+# - boot : Bootloader partition
+# - ESP-A : (500M) Kernel and initrd
+# - ESP-B : (500M)
+#
+# ZFS datasets do not necessary need to have specified size and can be
+# allocated dynamically. Quotas only restrict the maximum size of
+# datasets, but do not reserve the space in the pool.
+# The ZFS pool contains next datasets:
+# - root-A : (30G) Root FS
+# - root-B : (30G)
+# - vm-storage-A : (30G) Possible standalone pre-built VM images are stored here
+# - vm-storage-B : (30G)
+# - reserved-A : (10G) Reserved dataset, no use
+# - reserved-B : (10G)
+# - gp-storage : (50G) General purpose storage for some common insecure cases
+# - recovery : (no quota) Recovery factory image is stored here
+# - storagevm: (no quota) Dataset is meant to be used for StorageVM
+{pkgs, ...}: {
+  #TODO Probably the 'networking.hostId' should be set
+  # somewhere else instead.
+  networking.hostId = "8425e349";
+  disko = {
+    memSize = 4096;
+    extraPostVM = ''
+      ${pkgs.zstd}/bin/zstd --compress $out/*raw
+      rm $out/*raw
+    '';
+    extraRootModules = ["zfs"];
+    devices = {
+      disk.disk1 = {
+        type = "disk";
+        imageSize = "15G";
+        content = {
+          type = "gpt";
+          partitions = {
+            boot = {
+              name = "boot";
+              size = "1M";
+              type = "EF02";
+              priority = 1; # Needs to be first partition
```

Comment:
I made this a default now. But it also doesn't hurt to leave it: https://github.com/nix-community/disko/pull/688/files#diff-a91aff802e945d3aaba098fc7593c18276073bfc7877604a41c463498c50b0c2R53